.github/workflows/memory-profile-compare.yml

```diff
@@ -6,65 +6,41 @@ jobs:
   compare-memory-profiles:
     runs-on: ubuntu-latest
     steps:
-    - name: Checkout main branch
-      uses: actions/checkout@v2
+    - name: Checkout code
+      uses: actions/checkout@v3
       with:
-        ref: 'main'
-    
-    - name: Set up Go (main branch)
-      uses: actions/setup-go@v2
-      with:
-        go-version: '1.19'
-    
-    - name: Generate memory profile for main branch
-      run: make profile-memory OUTFILE_SUFFIX=.main
-      
-    - name: Save main branch memory profile
-      uses: actions/upload-artifact@v2
-      with:
-        name: memprofile-main
-        path: | 
-          memprofile.main.out
-          memprofile.main.text 
-    
-    - name: Checkout feature branch
-      uses: actions/checkout@v2
-      with:
-        ref: ${{ github.head_ref }}
-    
-    - name: Set up Go (feature branch)
-      uses: actions/setup-go@v2
-      with:
-        go-version: '1.19'
-    
-    - name: Generate memory profile for feature branch
-      run: make profile-memory OUTFILE_SUFFIX=.feat
-      
-    - name: Save feat branch memory profile
-      uses: actions/upload-artifact@v2
-      with:
-        name: memprofile-feat
-        path: | 
-          memprofile.feat.out
-          memprofile.feat.text 
+        fetch-depth: 0
 
-    - name: Download main branch memory profile
-      uses: actions/download-artifact@v2
+    - name: Set up Go
+      uses: actions/setup-go@v4
       with:
-        name: memprofile-main
-        path: .
-    
-    - name: Display structure of downloaded files
-      run: ls -R
-    
+        go-version: '1.21'
+
+    - name: Generate memory profiles
+      run: |
+        git checkout ${{ github.head_ref }}
+        make profile-memory OUTFILE_SUFFIX=.feat || { echo "Failed to generate feature profile"; ls -l; cat Makefile; exit 1; }
```

Comment:
Running the feature branch first so that I can demonstrate in this PR that its working.